src/tweenable.rs

```diff
@@ -894,264 +917,399 @@ mod tests {
-                    assert_eq!(cb_mon.invoke_count, times_completed as u64);
-                    assert_eq!(cb_mon.last_reported_count, times_completed);
-                    {
-                        let mut event_reader = event_reader_system_state.get_mut(&mut world);
-                        let event = event_reader.iter().next();
-                        if just_completed {
-                            assert!(event.is_some());
-                            if let Some(event) = event {
-                                assert_eq!(event.entity, dummy_entity);
-                                assert_eq!(event.user_data, USER_DATA);
-                            }
-                        } else {
-                            assert!(event.is_none());
-                        }
-                    }
-                }
+        let (mut world, _, mut event_writer) = create_event_reader_writer();
 
-                // Rewind
-                tween.rewind();
-                assert_eq!(tween.direction(), *tweening_direction); // does not change
-                assert!(abs_diff_eq(tween.progress(), 0., 1e-5));
-                assert_eq!(tween.times_completed(), 0);
-
-                // Dummy tick to update target
-                let actual_state = {
-                    let mut event_writer = event_writer_system_state.get_mut(&mut world);
-                    tween.tick(
-                        Duration::ZERO,
-                        &mut transform,
-                        Entity::from_raw(0),
-                        &mut event_writer,
-                    )
-                };
-                assert_eq!(actual_state, TweenState::Active);
-                let expected_translation = if tweening_direction.is_backward() {
-                    Vec3::ONE
-                } else {
-                    Vec3::ZERO
-                };
-                assert!(transform
+        tween.set_progress(starting_progress);
+        tween.tick(
+            Duration::ZERO,
+            &mut transform,
+            Entity::from_raw(0),
+            &mut event_writer.get_mut(&mut world),
+        );
+        // TODO(https://github.com/djeedai/bevy_tweening/issues/43) decide if this is acceptable behavior.
+        if starting_progress != 1. {
+            assert!(transform
+                .translation
+                .abs_diff_eq(Vec3::splat(starting_progress), 1e-5));
+        }
+
+        // Rewind
+        tween.rewind();
+        assert!(abs_diff_eq(tween.progress(), 0., 1e-5));
+        assert_eq!(
+            tween.tick(
+                Duration::ZERO,
+                &mut transform,
+                Entity::from_raw(0),
+                &mut event_writer.get_mut(&mut world),
+            ),
+            TweenState::Active
+        );
+        assert!(transform.translation.abs_diff_eq(Vec3::ZERO, 1e-5));
+        assert_eq!(tween.times_completed(), 0);
+    }
+
+    struct ExpectedValues<
```

Comment:
Ah yes agreed that's a bit ugly, but probably worth it for the understanding of the tests themselves. Thanks!

Reply:
Yup, agreed!